### PR TITLE
feat: `defaultTitle` override

### PR DESCRIPTION
**Pull request by jakobgrommas**
Switched from `descriptionFallback` to `defaultDescription`, added `defaultTitle`, modified `tryResolveQuery` to use regular expressions for more complex scenarios.

### CODE REVIEW

**Comment by johannschopplich** [comment]
Thanks for the PR! Can you please revert the version changes?

**Comment by grommasdietz** [comment]
Like this?

File: README.md [hunk 1]
```diff
@@ -49,6 +51,8 @@ By default, the SERP preview will render the title of the search engine result p
 2️⃣ **Title Separator** (defaults to `–`)
 3️⃣ **Site Title**
 
+When `defaultTitle` is defined
```
Comment (johannschopplich) [inline README.md:54]: Sentence is not finished.

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "kirby-serp-preview",
   "type": "module",
-  "version": "0.5.1",
+  "version": "0.5.2-beta.1",
```
Comment (johannschopplich) [inline package.json:4]: Can you please revert the version changes?

File: src/extensions/sections.php [hunk 3]
```diff
@@ -12,7 +12,8 @@
             'titleSeparator' => fn ($titleSeparator = '–') => $titleSeparator,
             'titleContentKey' => fn ($titleContentKey = null) => is_string($titleContentKey) ? strtolower($titleContentKey) : $titleContentKey,
             'descriptionContentKey' => fn ($descriptionContentKey = null) => is_string($descriptionContentKey) ? strtolower($descriptionContentKey) : $descriptionContentKey,
-            'descriptionFallback' => fn ($descriptionFallback = '') => $descriptionFallback,
+            'defaultTitle' => fn ($defaultTitle = '') => $defaultTitle,
+            'defaultDescription' => fn ($defaultDescription = '') => $defaultDescription,
```
Comment (johannschopplich) [inline src/extensions/sections.php:16]: I think we have to keep `descriptionFallback` as prop, don't we?

File: src/extensions/sections.php [hunk 4]
```diff
@@ -32,14 +33,24 @@
...
                 return $this->tryResolveQuery($this->titleSeparator);
             },
-            'descriptionFallback' => function () {
-                return $this->tryResolveQuery($this->descriptionFallback);
-            }
+            'defaultTitle' => function () {
+                return $this->tryResolveQuery($this->defaultTitle);
+            },
+            'defaultDescription' => function () {
+                return $this->tryResolveQuery($this->defaultDescription) ?: $this->tryResolveQuery($this->descriptionFallback);
```
Comment (johannschopplich) [inline src/extensions/sections.php:40]: Maybe simplify the approach. 

```suggestion
                return $this->tryResolveQuery($this->defaultDescription ?: $this->descriptionFallback);
```

File: src/extensions/sections.php [hunk 5]
```diff
@@ -32,14 +33,24 @@
...
+            },
+
         ],
         'methods' => [
             'tryResolveQuery' => function ($value, $fallback = null) {
-                if (is_string($value) && str_starts_with($value, '{{') && str_ends_with($value, '}}')) {
-                    return $this->model()->query(substr($value, 2, -2));
+                if (is_string($value)) {
```
Comment (johannschopplich) [inline src/extensions/sections.php:46]: The brackets have to be escaped, I believe:
```diff
- /{{(.+?)}}/
+ '/\{\{(.+?)\}\}/'
```

**Review by johannschopplich — CHANGES_REQUESTED** [review]
Some minor change requests. 🙌

**Comment by johannschopplich** [comment]
Thanks a lot for paving the road to v1.0. Much appreciated. 💚

**Pull request merged**